src/utils/todo/todo.js

```diff
@@ -0,0 +1,120 @@
+import { TodoFilter } from '../../utils/todo/todoFilter.js';
+
+import { getTodos, setTodos } from '../../service/todo.service.js';
+
+import { DragAndDrop } from '../dnd.js';
+import { reorderList } from '../../store/dnd.store.js';
+import { renderList, renderTodo, updateSummary } from '../../ui/todo/todo.render.js';
+import { changeFilterList } from '../../ui/todo/todo.dom.js';
+
+import {
+  changeCompleteTodo,
+  updateTodoState,
+  addTodo,
+  setTodoState,
+  increaseIncomplete,
+} from '../../store/todo.store.js';
+
+export class TodoListApp extends TodoFilter {
+  todoState = {
+    complete: 0,
+    incomplete: 0,
+  };
+
+  useDnd = false;
+
+  constructor({ id, wrapper, options }) {
+    super();
+
+    this.$wrapper = wrapper;
+    this.id = 'todo_' + id;
+    this.todos = getTodos(this.id);
+    setTodoState.bind(this)();
+    if (options?.useDnd) this.useDnd = true;
+  }
+
+  submitTodo() {
+    this.$form.addEventListener('submit', (e) => {
+      e.preventDefault();
+
+      const $target = e.target;
+      const $input = $target.querySelector('.todo-input');
+      const value = $input.value;
+
+      if (!value) {
+        alert('값을 입력하세요');
+        return;
+      }
+
+      $input.value = '';
+
+      increaseIncomplete.bind(this)();
+      addTodo.bind(this)(value);
+
+      if (this.filterState) {
+        updateSummary.bind(this)();
+        changeFilterList.bind(this)();
+      }
+    });
+  }
+
+  changeTodoState() {
+    this.$todo_body.addEventListener('change', (e) => {
+      const $target = e.target;
+
+      if ($target.closest('.base-checkbox')) {
+        const targetKey = Number($target.parentNode.children[0].id);
+        const targetChecked = $target.checked;
+        if (!targetKey) return;
+
+        changeCompleteTodo.bind(this)(targetKey);
+        renderList.bind(this)(this.todos);
+
+        if (!this.filterState) return;
+        updateTodoState.bind(this)(targetChecked);
+        this.updateFilter();
+
+        updateSummary.bind(this)();
```

Comment:
함수 실행에 관련된 변수 체크가 해당 함수 위에 있는게 좋을 것 같아서요!
```suggestion
        const $li = $target.closest('li');
        if (!$li) return;

        const targetKey = Number($target.closest('li').dataset.key);
        if (!targetKey) return;

        changeCompleteTodo.bind(this)(targetKey);
        renderList.bind(this)(this.todos);

        const targetChecked = $target.checked;
        updateTodoState.bind(this)(targetChecked);
        updateSummary.bind(this)();

        if (!this.filterState) return;
        this.updateFilter();
```